apps/aega/test/aega_SUITE.erl

```diff
@@ -484,7 +496,94 @@ oracle_extend(_Cfg) ->
     ok.
 
 %%%===================================================================
-%%% Oracle GA tests
+%%% Multi wrapping GA tests
+%%%===================================================================
+multi_wrap_spend(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    MinGP = aec_test_utils:min_gas_price(),
+    Acc1 = ?call(new_account, 1000000000 * MinGP),
+    Acc2 = ?call(new_account, 1000000000 * MinGP),
+    {ok, _} = ?call(attach, Acc1, "basic_auth", "authorize", []),
+    Auth = fun(N) -> #{ prep_fun => fun(Tx) -> ?call(basic_auth, Acc1, N, Tx) end } end,
+
+    SpendTx = aega_test_utils:spend_tx(#{sender_id    => aeser_id:create(account, Acc1),
+                                         recipient_id => aeser_id:create(account, Acc2),
+                                         amount       => 12345,
+                                         fee          => 20000 * MinGP,
+                                         nonce        => 0}),
+
+    %% First an incorrect inner "signing"
+    {_AuthData1, _InnerTx1, MetaTx1} = prep_meta(Acc1, Auth("444"), SpendTx),
+    {ok, #{tx_res := error}} =
+        ?call(meta, Acc1, Auth("1"), MetaTx1, #{}),
+
+    %% Now the happy case
+    {_AuthData2, _InnerTx2, MetaTx2} = prep_meta(Acc1, Auth("3"), SpendTx),
+    {ok, #{tx_res := ok}} =
+        ?call(meta, Acc1, Auth("2"), MetaTx2, #{}),
+
+    ok.
+
+multi_wrap_sc_create(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1 = ?call(new_account, 1000000000 * aec_test_utils:min_gas_price()),
+    Acc2 = ?call(new_account, 1000000000 * aec_test_utils:min_gas_price()),
+    {ok, _} = ?call(attach, Acc1, "basic_auth", "authorize", []),
+    {ok, _} = ?call(attach, Acc2, "basic_auth", "authorize", []),
+    Auth1 = fun(N) -> #{ prep_fun => fun(Tx) -> ?call(basic_auth, Acc1, N, Tx) end } end,
+    Auth2 = fun(N) -> #{ prep_fun => fun(Tx) -> ?call(basic_auth, Acc2, N, Tx) end } end,
+
+    Amnt = 1000000 * aec_test_utils:min_gas_price(),
+    OffState = aega_test_utils:new_state(Acc1, Amnt, {basic, "2"}, Acc2, Amnt, {basic, "2"}),
+    {ok, #{tx_res := ok}} =
+        ?call(ga_channel_create, Acc1, Auth1("1"), Acc2, Auth2("1"), OffState, #{}),
+
+    ok.
+
+multi_wrap_sc_close_mutual(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1 = ?call(new_account, 1000000000 * aec_test_utils:min_gas_price()),
+    Acc2 = ?call(new_account, 1000000000 * aec_test_utils:min_gas_price()),
+    {ok, _} = ?call(attach, Acc1, "basic_auth", "authorize", []),
+    {ok, _} = ?call(attach, Acc2, "basic_auth", "authorize", []),
+    Auth1 = fun(N) -> #{ prep_fun => fun(Tx) -> ?call(basic_auth, Acc1, N, Tx) end } end,
```

Comment:
could be a utility function taking Account as parameter and returning this function: Auth1 = auth_account(Acc1),